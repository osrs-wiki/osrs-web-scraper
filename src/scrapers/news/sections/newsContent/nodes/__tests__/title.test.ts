import { MediaWikiBuilder } from "@osrs-wiki/mediawiki-builder";
import parse from "node-html-parser";

import titleParser from "../title";

describe("title node", () => {
  test("Smalltitle tag should parse and render as level 3 header", () => {
    const root = parse("<smalltitle>Test</smalltitle>");
    const builder = new MediaWikiBuilder();
    builder.addContents([titleParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("Bigtitle tag should parse and render as level 2 header", () => {
    const root = parse("<bigtitle>Test</bigtitle>");
    const builder = new MediaWikiBuilder();
    builder.addContents([titleParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("Smalltitle with complex content should parse correctly", () => {
    const root = parse("<smalltitle>Test with <b>bold</b> text</smalltitle>");
    const builder = new MediaWikiBuilder();
    builder.addContents([titleParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("Bigtitle with complex content should parse correctly", () => {
    const root = parse("<bigtitle>Test with <i>italic</i> text</bigtitle>");
    const builder = new MediaWikiBuilder();
    builder.addContents([titleParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });
});