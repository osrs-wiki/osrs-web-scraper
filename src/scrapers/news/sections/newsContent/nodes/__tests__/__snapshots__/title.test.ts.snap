// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`title node Bigtitle tag should parse and render as level 2 header 1`] = `"==Test=="`;

exports[`title node Bigtitle with complex content should parse correctly 1`] = `"==Test with ''italic'' text=="`;

exports[`title node Smalltitle tag should parse and render as level 3 header 1`] = `"===Test==="`;

exports[`title node Smalltitle with complex content should parse correctly 1`] = `"===Test with '''bold''' text==="`;
